aq_zombie_finder/aq_zombie_finder.py

```diff
@@ -0,0 +1,186 @@
+        :param aquilon_zombie_filepath: The filepath of the Aquilon output file (String)
+    """
+    # Define the Regex expression used to find the Serial number
+    serial_rexp = compile(r"((?<=Serial: )(.+?)(?=\\r))")
+
+    # Find the serial in the host information using Regex
+    serial = serial_rexp.findall(str(aq_host))[0][0]
+
+    # Get the host assigned to that serial in Openstack
+    server_details = ssh_command(
+        openstack_client, "source admin-openrc.sh ; openstack server show " + serial
+    )
+
+    if "No server with a name or ID of" in str(server_details):
+        # If the serial doesn't return a host, save its serial and IP as a potential zombie
+        with open(aquilon_zombie_filepath, "a") as aquilon_zombie_file:
+            aquilon_zombie_file.write(serial + " | " + aq_ip + "\n")
+
+
+def check_output_files(openstack_zombie_filepath, aquilon_zombie_filepath):
+    """
+    Function to check if output files already exist
+        :param aquilon_zombie_filepath: The filepath of the Aquilon output file (String)
+        :param openstack_zombie_filepath: The filepath of the Openstack output file (String)
+    """
+    if os.path.exists(openstack_zombie_filepath):
+        raise RuntimeError(f"{openstack_zombie_filepath} already exists")
+    if os.path.exists(aquilon_zombie_filepath):
+        raise RuntimeError(f"{aquilon_zombie_filepath} already exists")
+
+
+def aq_zombie_finder():
+    # Get arguments passed to the script
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        "-u", "--user", metavar="USER", help="FedID of the user", required=True
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        metavar="PASSWORD",
+        help="Password of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-i",
+        "--ip",
+        metavar="IP",
+        help="IP of the machine with Openstack to SSH to",
+        required=True,
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="OUTPUT",
+        help="Directory to create the output files in",
+    )
+
+    # Define the variables with the script arguments
+    args = parser.parse_args()
+    user = args.user
+    password = args.password
+    openstack_ip = args.ip
+    output = args.output
+
+    # Create a paramiko SSH client to the VM running Openstack and to Aquilon
+    openstack_client = create_client(openstack_ip, user, password)
+    aquilon_client = create_client("aquilon.gridpp.rl.ac.uk", user, password)
+
+    # Define the filepath for the output to be saved to
+    openstack_zombie_filepath = os.path.join(
+        output or "output", "openstack_zombie_list.txt"
+    )
+    aquilon_zombie_filepath = os.path.join(
+        output or "output", "aquilon_zombie_list.txt"
+    )
+    check_output_files(openstack_zombie_filepath, aquilon_zombie_filepath)
```

Comment:
you can remove this function and just do the validation check with a for loop - see #78 for the same thing